src/lib/oauth.ts

```diff
@@ -94,11 +95,12 @@ export async function handleAuthCallback( hash: string ): Promise< Error | Store
 	};
 }
 
-export function authenticate(): void {
+export async function authenticate(): Promise< void > {
 	const authUrl = `${ WP_AUTHORIZE_ENDPOINT }?response_type=token&client_id=${ CLIENT_ID }&redirect_uri=${ encodeURIComponent(
 		REDIRECT_URI
 	) }&scope=${ encodeURIComponent( SCOPES ) }`;
-	shell.openExternal( authUrl );
+
+	await shellOpenExternalWrapper( authUrl );
```

Comment:
Is this a leftover? `shellOpenExternalWrapper` isn't async

Reply:
No, it's async. Moreover it's key thing to catch the error, since `shell.openExternal` is async.

Reply:
UPDATE: With my latest changes we don't need it